rusk-wallet/src/wallet/address.rs

```diff
@@ -26,10 +27,13 @@ pub enum Address {
 
 impl Address {
     /// Check if the `other` Address uses the same transaction model
+    ///
+    /// # Errors
+    /// If the other address is not the same type as `Self`.
     pub fn same_transaction_model(&self, other: &Address) -> Result<(), Error> {
         match (self, other) {
-            (Address::Shielded(_), Address::Shielded(_)) => Ok(()),
-            (Address::Public(_), Address::Public(_)) => Ok(()),
+            (Address::Shielded(_), Address::Shielded(_))
+            | (Address::Public(_), Address::Public(_)) => Ok(()),
```

Comment:
This is a matter of preference
I personally find the previous version more clear

Reply:
prefer so much that you want me to add an exception for it?